File: backend/src/main/java/com/aiemail/generator/template/service/TemplateService.java
package com.aiemail.generator.template.service;

import com.aiemail.generator.auth.entity.UserEntity;
import com.aiemail.generator.template.dto.TemplateRequest;
import com.aiemail.generator.template.dto.TemplateResponse;
import org.springframework.data.domain.Page;

import java.util.UUID;

public interface TemplateService {
    TemplateResponse createTemplate(TemplateRequest request, UserEntity user);
    TemplateResponse updateTemplate(UUID id, TemplateRequest request, UserEntity user);
    void deleteTemplate(UUID id, UserEntity user);
    TemplateResponse getTemplateById(UUID id, UserEntity user);
    Page<TemplateResponse> getTemplates(UserEntity user, String name, com.aiemail.generator.common.enums.Tone tone, String category, int page, int size, String sortBy);
}

File: frontend/src/services/templateService.js
import api from './api';

export const templateService = {
  getLibrary: async () => {
    const response = await api.get('/api/v1/templates/library');
    return response.data;
  },

  getTemplates: async (params) => {
    const response = await api.get('/api/v1/templates', { params });
    return response.data;
  },

  createTemplate: async (data) => {
    const response = await api.post('/api/v1/templates', data);
    return response.data;
  },

  updateTemplate: async (id, data) => {
    const response = await api.put(`/api/v1/templates/${id}`, data);
    return response.data;
  },

  deleteTemplate: async (id) => {
    const response = await api.delete(`/api/v1/templates/${id}`);
    return response.data;
  }
};

export default templateService;
